### PR TITLE
PRIME-2320 add oral health professionals

### PR DESCRIPTION
**Pull request by bergomi02**
https://pr-2351.pharmanetenrolment.gov.bc.ca/

Note: Test Card 1 and 2 are enrolled with old license classes.

### CODE REVIEW

File: prime-dotnet-webapi/Migrations/20230609220819_AddSocialWorkers.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace Prime.Migrations
+{
+    public partial class AddSocialWorkers : Migration
```
Comment (neophyte57) [inline prime-dotnet-webapi/Migrations/20230609220819_AddSocialWorkers.cs:6]: Somehow this data migration snuck into this branch?

Comment (bergomi02) [reply prime-dotnet-webapi/Migrations/20230609220819_AddSocialWorkers.cs:6]: I added the social worker migration into this branch so the ID of the license class will align properly. The only drawback is the social worker change need to go first. 
However, I noticed the migration for Oral Health Professional is not executed in PR. I am currently investigating that.

File: prime-angular-frontend/src/app/shared/components/forms/college-certification-form/college-certification-form.component.ts [hunk 2]
```diff
@@ -382,4 +398,25 @@ export class CollegeCertificationFormComponent implements OnInit {
   private filterPractices(collegeCode: number): PracticeConfig[] {
     return this.practices.filter(p => p.collegePractices.map(cl => cl.collegeCode).includes(collegeCode));
   }
+
+  private checkLicenseIfDiscontinued() {
+    if (this.collegeCode.value && this.licenseCode.value) {
+      this.licenseClassDiscontinued = this.isCertificationDiscontinued(this.collegeCode.value, this.licenseCode.value);
+      this.collegeLicenseName = `${this.configService.colleges.find(c => c.code === this.collegeCode.value).name} -` +
```
[the remark this answers is not in the document]
Comment (bergomi02) [reply prime-angular-frontend/src/app/shared/components/forms/college-certification-form/college-certification-form.component.ts]: This is fixed.

File: prime-dotnet-webapi/Configuration/Database/Seeded Tables/CollegeConfiguration.cs [hunk 3]
```diff
@@ -10,27 +10,27 @@ public override IEnumerable<College> SeedData
...
+                    new College { Code = 14, Name = "College of Optometrists of BC", Weight = 120                                                  },
+                    new College { Code = 15, Name = "College of Physical Therapists of BC", Weight = 130                                           },
+                    new College { Code = 16, Name = "College of Psychologists of BC", Weight = 140                                                 },
+                    new College { Code = 17, Name = "College of Speech and Hearing Health Professionals of BC", Weight = 160                       },
+                    new College { Code = 18, Name = "College of Traditional Chinese Medicine Practitioners and Acupuncturists of BC", Weight = 170 },
+                    new College { Code = 19, Name = "BC College of Social Workers", Weight = 180                                                   },
+                    new College { Code = 20, Name = "College of Oral Health Professionals", Weight = 50                                            }
```
Comment (neophyte57) [inline prime-dotnet-webapi/Configuration/Database/Seeded Tables/CollegeConfiguration.cs:33]: Considering that there are 4 enrollments from College of Chiropractors vs. 125 Production licences from the four previous dental colleges, College of Oral Health Professionals should be listed higher
image

File: prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseDetailConfiguration.cs [hunk 4]
```diff
@@ -292,6 +293,23 @@ public override IEnumerable<LicenseDetail> SeedData
...
+
+                    //College of Oral Health Professionals
+                    new LicenseDetail { Id =  238, LicenseCode = 87, Prefix = "", Manual = false, Validate = false, LicensedToProvideCare = true, NamedInImReg = false, PrescriberIdType = null, AllowRequestRemoteAccess = false, EffectiveDate = June13th2023, CreatedTimeStamp = SEEDING_DATE, UpdatedTimeStamp = SEEDING_DATE },
+                    new LicenseDetail { Id =  239, LicenseCode = 88, Prefix = "", Manual = false, Validate = false, LicensedToProvideCare = true, NamedInImReg = false, PrescriberIdType = null, AllowRequestRemoteAccess = false, EffectiveDate = June13th2023, CreatedTimeStamp = SEEDING_DATE, UpdatedTimeStamp = SEEDING_DATE },
+                    new LicenseDetail { Id =  240, LicenseCode = 89, Prefix = "", Manual = false, Validate = true, LicensedToProvideCare = true, NamedInImReg = true, PrescriberIdType = PrescriberIdType.Mandatory, AllowRequestRemoteAccess = false, EffectiveDate = June13th2023, CreatedTimeStamp = SEEDING_DATE, UpdatedTimeStamp = SEEDING_DATE },
```
Comment (neophyte57) [inline prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseDetailConfiguration.cs:300]: `PractId`/`PrescriberIdType` is only applicable for BCCNM

**Comment by neophyte57** [comment]
Getting this error when trying to accept TOA when submission specifies a licence class for discontinued College of Dental Hygenists of BC

image

image

**Comment by bergomi02** [comment]
> Getting this error when trying to accept TOA when submission specifies a licence class for discontinued College of Dental Hygenists of BC
> 
> image
> 
> image

Can't reproduce the error in PR env and local? Not sure what causing it. 
FYI, because the database migration has been modified couple times and other DB changes have been merged from Production release, I have to create a new PR to create a new migration from scratch for the college order change and the license detail update. Please see PR-2363. Please leave this PR open as reference, will close this PR after the implementation merges to Dev.

**Comment by bergomi02** [comment]
The implementation has been reworked to another other PR. closing this PR.

**Pull request closed without merging**